docs/README.md

```diff
@@ -1,112 +1,132 @@
-# How proxless works
-  
-> Proxless looks for any kubernetes services with specific annotations and make their associated deployment serverless.
- 
-It works with 2 modes:
+# How works proxless
+
+## Wording
+
+- `service` == `kubernetes service`
+- `deployment` == `kubernetes deployment`
+
+## In 1 minute
+
+The main idea when building proxless is to be as simple as possible.  
+You don't wanna install a Custom Resource Definition, you don't want to have a big stack, you just want something to scale up and down your deployment based on-demand.  
+Proxless is a simple proxy written in golang and does not consume much resources. You don't need anything else.
+
+You just need to add an annotation on the services you want to make on-demand and Proxless will handle the rest.
+
+Start now and deploy the [helm chart](../helm/README.md).  
 
-- **Namespace scoped** - env var `NAMESPACE` is provided - proxless will only look for services within this namespace.
-- **Global** - env var `NAMESPACE` is empty - proxless will look for any services in the cluster.
+### Examples
 
+Check the [example](../examples). Replace `hello-world.proxless.kintohub.net` with your own domain for testing.  
+_Notes: don't forget to configure your DNS_
 
-## ServicesEngine
+## Namespace scoped or cluster wide
+ 
+- **Namespace scoped**
+    - env var `NAMESPACE` must not be empty - proxless will only look for services within this namespace.
+    - a `Role` is required.  See [here](../helm/templates/role.yaml).
+- **Cluster wide**
+    - env var `NAMESPACE` is empty - proxless will look for any services in the cluster.
+    - a `ClusterRole` is required. See [here](../helm/templates/clusterrole.yaml).
+
+## Core Concepts
 
-The ServiceEngine is accessible [here](../internal/kubernetes/servicesengine).  
-It run as a separate routine. It is responsible for looking at the services and storing the information in the store.  
+Proxless works with 4 core concepts:
 
-### Get services
+- the **store** is an in-memory storage containing the information for each route.
+- the **proxy**, responsible for forwarding the requests to the correct back end and scaling up the deployment if needed.
+- the **services engine**, responsible for retrieving the configuration from the services and configuring the deployments.
+- the **downscaler**, responsible for downscaling the deployments when they are not used.
 
-It will look for any kubernetes services with the 2 annotations
+### The store
```

Comment:
Comment - Maybe a future in-memory store for scaling out, reed about it a long time ago and never used it but was always curious about it: https://github.com/golang/groupcache

Reply:
This looks cool.
Especially after this comment -> https://stackoverflow.com/questions/50486223/how-to-configure-a-distributed-hashtable-on-kubernetes
Don't really know how but it seems possible to get all the IP of all the replicas using a DNS lookup.

Reply:
I checked, don't think it is what we think -> https://capotej.com/blog/2013/07/28/playing-with-groupcache/

> Another important consideration is that theres no support for set/update/evict operations, all you get is GET. Really fast, consistent, distributed GET’s.

> Once you realize that groupcache is more of a smart, distributed LRU cache, rather than an outright memcached replacement, it all makes much more sense. Especially considering what it was built for, caching immutable file blobs for dl.google.com.